### PR TITLE
Add user and example docker-compose

### LINKED ISSUE #3: User Permission Errors On Linux
If you attempt to run the container as-is, you will get a permission error on Linux (tested on Arch, Ubuntu, and CentOS). This is expected, however, as there is no default UUID and GUID set in the Dockerfile, and wasn't explained in a docker-compose.yml. I propose to add `USER 1000:1000` to the Dockerfile, and include an example `docker-compose.yml` for those interested in using that. I've made a PR with these changes.

### PR DESCRIPTION
**Pull request by TheCatster**
Proposed solution to user permission problems on Linux.

Closes #3

### CODE REVIEW

**Comment by sethmachine** [comment]
Awesome, I appreciate the pull request for fixing this issue.  There are a lot of similar repos on GitHub for running a Valheim server on Docker.  What I hope separates this one is the focus is on being educational, so it's friendly to hobbyists.  What I don't want to offer is just a "pull, run, and forget" attitude for this Docker image.  

With that in mind, before your PR is merged, could you update the readme to include instructions on when, how and why to run the docker-compose.yml as opposed to just running a Docker image?

File: Dockerfile [hunk 1]
```diff
@@ -9,6 +9,8 @@ RUN ./steamcmd.sh +login anonymous \
 +app_update 896660 \
 validate +exit
 
+USER 1000:1000
```
Comment (sethmachine) [inline Dockerfile:13]: 1.  What is the issue with using the `steam` user that comes from the `steamcmd2` base image?  
1.  Mind adding a comment here explaining what this `USER 1000:1000` line does?

**Comment by TheCatster** [comment]
Absolutely! I personally use this container alongside a larger one anyways, but I just wanted to help out. I'll add those details.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+version: "3"
+services:
+  valheim:
+    image: sethmachineio/valheim-server
+    user: "1000:1000"
```
Comment (sethmachine) [inline docker-compose.yml:5]: nit, same as prior, what is the issue with having the user be `steam` (from the base image) ?

File: Dockerfile (hunk 1 above)
Comment (TheCatster) [reply Dockerfile:13]: It doesn't have the same IDs, so there is an ownership conflict with folders unless a steam user is also created on the server (or whatever machine). The USER variable represents the UUID and GUIDs, and I can add that comment.

**Comment by TheCatster** [comment]
You said you run this on a Mac using Docker Desktop, right? Maybe that's the disconnect. I've never used desktop docker.

**Comment by sethmachine** [comment]
Exactly, yes I've only tested this with Docker Desktop on macOS.  That may be why Linux issues like permissions have been popping up.  

And mind confirming that the new image works as expected and writing this in your PR description?  I've done a manual test where I build a local image, run a brand new world and then stop it.  I then check to see if the `.db` file is correctly generated.  This can be automated in the future.

**Comment by TheCatster** [comment]
The image I had worked fine, I'd been running a server for friends using it. Adding the comments won't (I sure hope!) change that. And yeah, permissions are a fairly common issue that needs to be addressed, since otherwise the user on the GNU/Linux OS (think like your user) won't be able to access/edit the files within. Or, the container won't be able to write to the dir.

**Comment by TheCatster** [comment]
Do you want a comment in the docker compose as well, or just in the Dockerfile and README?

**Comment by TheCatster** [comment]
Added documentation, let me know what you think.

**Review by sethmachine — APPROVED** [review]

**Pull request merged**